### PR TITLE
Customized colors.

### LINKED ISSUE #21: dark mode

### PR DESCRIPTION
**Pull request by xiaoyu2006**
# New: customized colors

Fix #21 

**Screenshot after this change**
截屏2019-11-19下午8 15 33

### CODE REVIEW

**Comment by timqian** [comment]
Hi, @xiaoyu2006 Thanks for this PR.

I think setting the chart background to 'black' and 'white' maybe not be good in some situations. Sometimes people may have their own dark/light background-color. If their background-color is not black or white, the chart may not fit in.

To make chart.xkcd look good on dark background, I am thinking, maybe an option to control the stroke color will be enough.

For example, people can add an option

`strokeColor: 'white'` to make the lines/words white

So that the chart looks better in dark background

**Comment by xiaoyu2006** [comment]
So what about adding the following options instead of `is_dark_style`:
```javascript
options = {
// -- snip --
strokeColor: 'black',
backgroundColor: 'white',
legendColor: 'white',
}
```

**Comment by xiaoyu2006** [comment]
It's ok now.

File: docs/02-getting-started.md [hunk 1]
```diff
@@ -33,3 +36,12 @@ In the following example we create a line chart.
 - `yLabel`: optional y label of the chart
 - `data`: the data you want to visulize
 - `options`: optional configurations to customize how the chart looks
+
+## Customize colors
+
+*Because when you use `options: {...}` you are actually assigning a value to it and it will override all of the default values. While the following opts can't be `undefined`, I use class constructors to build them.*
```
Comment (xiaoyu2006) [inline docs/02-getting-started.md]: Is it the only way to do this?

Comment (xiaoyu2006) [reply docs/02-getting-started.md]: I figured it out.

**Comment by xiaoyu2006** [comment]
I guess I shouldn't treat Git so forcefully. `master branch` works well, but all the commits are confusing.

**Comment by timqian** [comment]
> So what about adding the following options instead of `is_dark_style`:
```javascript
options = {
// -- snip --
strokeColor: 'black',
backgroundColor: 'white',
legendColor: 'white',
}
```
About the options you suggest, I think

`strokeColor` and `backgroundColor` is nice to have. 

The `legendColor` is a little confusing as the user can not sure if this color denotes the color of legend texts or background.

After reading your code, I know `legendColor` means legend Background color.

 I think removing the `legnedColor` and use `backgroundColor` as the background of legend should be fine. It is less confusing and simpler.

So the added options will be 

```javascript
options = {
// -- snip --
strokeColor: 'black',
backgroundColor: 'white',
}
```

File: src/components/Tooltip.js [hunk 2]
```diff
@@ -24,22 +24,24 @@ class Tooltip {
...
       .attr('y', this._getUpLeftY())
       .style('visibility', 'hidden');
 
     this.tipBackground = this.svg.append('rect')
-      .style('fill', 'white')
-      .attr('fill-opacity', 0.85)
-      .attr('stroke', '#aaa')
+      .style('fill', this.legendColor)
+      .attr('fill-opacity', .9)
+      .attr('stroke', this.strokeColor)
```
Comment (timqian) [inline src/components/Tooltip.js]: border of the tooltip better be not exactly the same as legend. But not sure what is a good way

**Comment by xiaoyu2006** [comment]
How about now?

**Comment by xiaoyu2006** [comment]
Snapshot:
截屏2019-11-20下午9 48 31

File: src/Bar.js [hunk 3]
```diff
@@ -69,16 +82,18 @@ class Bar {
...
+    if (this.xLabel) addLabels.xLabel(this.svgEl, this.xLabel, this.strokeColor);
+    if (this.yLabel) addLabels.yLabel(this.svgEl, this.yLabel, this.strokeColor);
 
     const tooltip = new Tooltip({
       parent: this.svgEl,
       title: 'tooltip',
       items: [{ color: 'red', text: 'weweyang: 12' }, { color: 'blue', text: 'timqian: 13' }],
       position: { x: 30, y: 30, type: config.positionType.upRight },
       unxkcdify: this.options.unxkcdify,
+      legendColor: this.legendColor,
```
Comment (timqian) [inline src/Bar.js]: how about 
```
backgroundColor: this.backgroundColor
```

File: src/Line.js [hunk 4]
```diff
@@ -46,6 +55,9 @@ class Line {
       datasets,
     };
     this.options = options;
+    this.strokeColor = options.strokeColor;
+    this.backgroundColor = options.backgroundColor;
+    this.legendColor = options.legendColor;
```
Comment (timqian) [inline src/Line.js]: similar issue

File: src/Radar.js [hunk 5]
```diff
@@ -71,6 +84,8 @@ class Radar {
       items: [],
       position: { x: 0, y: 0, type: config.positionType.downRight },
       unxkcdify: this.options.unxkcdify,
+      strokeColor: this.strokeColor,
+      legendColor: this.legendColor,
```
Comment (timqian) [inline src/Radar.js]: backgroundColor

File: src/Radar.js [hunk 6]
```diff
@@ -232,6 +249,8 @@ class Radar {
         unxkcdify: this.options.unxkcdify,
         parentWidth: this.width,
         parentHeight: this.height,
+        legendColor: this.legendColor,
```
Comment (timqian) [inline src/Radar.js]: backgroundColor

File: src/utils/addLegend.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import config from '../config';
 
 export default function addLegend(parent, {
-  items, position, unxkcdify, parentWidth, parentHeight,
+  items, position, unxkcdify, parentWidth, parentHeight, strokeColor, legendColor,
```
Comment (timqian) [inline src/utils/addLegend.js]: backgroundColor

**Comment by timqian** [comment]
This PR looks good to me expect
1. the internal state `legendColor` seems not necessary to me
2. border of the tooltip better be not exactly the same as legend. But not sure what is a good way, so I think I will merge this PR after you remove the `legendColor` from the code and test locally to find a good way to handle it

**Comment by xiaoyu2006** [comment]
Done!

**Pull request merged**